crates/ecs/src/profiling.rs

```diff
@@ -19,10 +19,15 @@ pub enum ProfilingError {
 /// Whether a profiling operation succeeded.
 pub type ProfilingResult<T, E = ProfilingError> = Result<T, E>;
 
-impl Application {
+/// Represents an [`Application`] which can be profiled.
+pub trait Profileable: Sized {
     /// Enables profiling connection to Tracy.
+    fn with_profiling(self) -> ProfilingResult<Self>;
```

Comment:
Does this pattern have a name? I'm really enjoying it 😏

Reply:
I'm not sure if it has its own name, but it's inspired by how builders work!